ledger/eval.go

```diff
@@ -728,15 +728,19 @@ func (eval *BlockEvaluator) transactionGroup(txgroup []transactions.SignedTxnWit
 	cow := eval.state.child(len(txgroup))
 
 	// Prepare eval params for any ApplicationCall transactions in the group
-	evalParams := eval.prepareEvalParams(txgroup)
+	spec := transactions.SpecialAddresses{
```

Comment:
it would be cleaner if you'll move the `transactions.SpecialAddresses` to be a member of the `BlockEvaluator`. That way, you won't need to pass it to `eval.transaction`, and would still have it available for `eval.prepareEvalParams`. As a bonus, it would also allow you to cleanup `testTransaction` and use it there as well.

Reply:
Good idea, since that also means avoiding many more allocations.